projects/plugins/jetpack/extensions/blocks/nextdoor/nextdoor.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Nextdoor Block.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Nextdoor;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'nextdoor';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	Blocks::jetpack_register_block(
+		BLOCK_NAME,
+		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Nextdoor block registration/dependency declaration.
+ *
+ * @param array $attr    Array containing the Nextdoor block attributes.
+ * @return string
+ */
+function load_assets( $attr ) {
+	$pattern = '/^http[s]?:\/\/((?:www\.)?nextdoor(?:.*)?\/(?:embed)\/[a-z0-9\/\?=_\-\.\,&%$#\@\!\+]*)/i';
+
+	$url = isset( $attr['url'] ) ? $attr['url'] : null;
+
+	if ( empty( $url ) ) {
+		return;
+	}
+
+	if ( ! preg_match( $pattern, $url ) ) {
```

Comment:
What do you think about switching to `Jetpack_Gutenberg::validate_block_embed_url()` here, for extra safetly?

Reply:
Addressed in https://github.com/Automattic/jetpack/pull/33751/commits/4d78df79aeac2f9b8b4b9abb19603d1d2520429f